index.js

```diff
@@ -1,255 +1,50 @@
-            , 'user-agent': config.UserAgent
-        }, timeout: MAX_REQUEST_TIMEOUT_MS
-        , withCredentials: false
-        , agent: new agent.HttpsAgent({
-            freeSocketTimeout: FREE_SOCKET_TIMEOUT_MS
-        })
-    };
-
-    // Flush the Log
-    async.retry({
-        times: MAX_REQUEST_RETRIES
-        , interval: retryCount => REQUEST_RETRY_INTERVAL_MS * Math.pow(2, retryCount)
-        , errorFilter: errCode => DEFAULT_HTTP_ERRORS.includes(errCode) || errCode === 'INTERNAL_SERVER_ERROR'
-    }, (reqCallback) => {
-        return request(options, (error, response, body) => {
-            if (error) return reqCallback(error.code);
-            if (response.statusCode >= INTERNAL_SERVER_ERROR) return reqCallback('INTERNAL_SERVER_ERROR');
-            return reqCallback(null, body);
-        });
-    }, callback);
-};
-
-// Main Handler
-exports.handler = (event, context, callback) => {
-    const config = getConfig(event)
-        , eventData = processEvent(parseEvent(event))
-        , payload = config.eventlog ? [prepareEvent(eventData)] : []
-        , s3params = {
-            Bucket: eventData && eventData.meta && eventData.meta.bucket && eventData.meta.bucket.name || undefined
-            , Key: eventData && eventData.meta && eventData.meta.object && eventData.meta.object.key || undefined
-        };
-
-    if (config.filelog) {
-        return getLogs(s3params, (error, lines) => {
-            if (error) {
-                if (config.eventlog) return send(payload, config, callback);
-                return callback(error);
-            }
-
-            return async.everySeries(batchify(payload.concat(prepareLogs(lines, eventData))), (batch, next) => {
-                setTimeout(() => {
-                    return send(batch, config, next);
-                }, BATCH_INTERVAL_MS);
-            }, callback);
-        });
-    } else if (!config.eventlog) {
-        return callback('None of file and event logging has been enabled!');
-    }
-
-    return send(payload, config, callback);
-};
+'use strict'
+
+const async = require('async')
+
+const config = require('./lib/config.js')
+const {handleEvent} = require('./lib/event-handler.js')
+const {flush} = require('./lib/logger.js')
+const {getLogs, prepareLogs} = require('./lib/transformer.js')
+const {batchify, getProperty} = require('./lib/utils.js')
+
+const DOT_REGEXP = /\./g
+
+module.exports = {
+  handler
+}
+
+async function handler(event, context, callback) {
+  config.validateEnvVars()
+  const tags = config.get('tags')
+  if (tags) {
+    config.set('tags', tags.split(',').map((tag) => {
+      return tag.trim()
+    }).join(','))
+  }
+
+  const eventData = handleEvent(event)
+  const s3params = {
+    Bucket: getProperty(eventData, 'meta.bucket.name')
+  , Key: getProperty(eventData, 'meta.object.key')
+  }
+
+  let lines
+  try {
+    lines = getLogs(s3params)
+  } catch (e) {
+    return callback(e)
+  }
```

Comment:
an `async` handler should not use a callback, just let it throw. [reference](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html#nodejs-handler-async)

Reply:
Done!